plugin/loader/loader.go

```diff
@@ -140,18 +141,22 @@ func (loader *PluginLoader) Load(pl plugin.Plugin) error {
 	}
 
 	name := pl.Name()
-	if ppl, ok := loader.plugins[name]; ok {
-		// plugin is already loaded
-		return fmt.Errorf(
-			"plugin: %s, is duplicated in version: %s, "+
-				"while trying to load dynamically: %s",
-			name, ppl.Version(), pl.Version())
+
+	for _, p := range loader.plugins {
```

Comment:
`O(n**2)`, you could use a `map[string]struct{}` to keep of already registered plugins

Reply:
Future madman that load enough plugin for that to be an issue, you are mad. Also, please ping this issue and explain why, I'm so curious.